picard/ui/util.py

```diff
@@ -81,15 +81,71 @@ def find_starting_directory():
     return find_existing_path(path)
 
 
-class MultiDirsSelectDialog(QtWidgets.QFileDialog):
+def picardize_caption(caption):
+    return _("Picard - %s") % caption
+
+
+def convert_filedialog_params(kwargs, default_caption):
+    # We use PySide/Qt syntax in code, so convert to PyQt6
+    if 'dir' in kwargs:
+        # TODO: remove with PySide
+        # pyside & qt use `dir`, pyqt uses `directory`
+        kwargs['directory'] = kwargs['dir']
+        del kwargs['dir']
+
+    if 'selectedFilter' in kwargs:
+        # TODO: remove with PySide
+        # pyside & qt use `selectedFilter`, pyqt uses `initialFilter`
+        kwargs['initialFilter'] = kwargs['selectedFilter']
+        del kwargs['selectedFilter']
+
+    # Ensure we have a caption in those dialogs
+    if 'caption' not in kwargs or not kwargs['caption']:
+        kwargs['caption'] = default_caption
+
+    # Add app name to the caption, fancier in window managers
+    kwargs['caption'] = picardize_caption(kwargs['caption'])
+
+
+class FileDialog(QtWidgets.QFileDialog):
+    """Wrap QFileDialog & its static methods"""
+
+    def __init__(self, parent=None, caption="", directory="", filter=""):
+        if not caption:
+            caption = _("Select a file or a directory")
+        caption = picardize_caption(caption)
+        super().__init__(parent=parent, caption=caption, directory=directory, filter=filter)
+
+    @staticmethod
+    def getSaveFileName(**kwargs):
+        convert_filedialog_params(kwargs, _("Select a target file"))
+        return QtWidgets.QFileDialog.getSaveFileName(**kwargs)
```

Comment:
Just something we need to be aware of with this implementation: `FileDialog.getSaveFileName` gives the caller a `QFileDialog`, not a `FileDialog`.

This might become unexpected should we for example replace existing uses of `QFileDialog` with `FileDialog` and add additional default handling to this class.

We currently have one direct instantiation of `QFileDialog` at https://github.com/metabrainz/picard/blob/master/picard/tagger.py#L1204 . Probably this should also be changed to the static helper functions instead?

I think I avoided the helper functions when implementing this because I found the many arguments to a single static function harder to use then just instantiating the class itself and setting properties on it.

Another thing is especially with these functions with many parameters but only using `**kwargs` that I need to lookup the possible parameters and code completion cannot assist me. Currently I would get something like this:

![grafik](https://github.com/metabrainz/picard/assets/29852/4639203a-2f93-4189-9176-ee54d5bdf1ad)

That would definitely motivate me to avoid the wrapper and use the class directly. Maybe we should just have our very own specific wrappers that open the dialog as we actually need it right now so we can easily use it and have consistent behavior.

Another point is how we dealt with the separate `MultiDirsSelectDialog`. Should we make this a subclass of `FileDialog`? We should also use consistent caption there at least.

Reply:
I moved those changes outside this branch, I'll submit a PR after this one is merged.